src/main/java/org/usfirst/frc/team2473/robot/subsystems/SparkDriveSubsystem.java

```diff
@@ -0,0 +1,250 @@
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import org.usfirst.frc.team2473.framework.CHS_SparkMax;
+
+/**
+ * This class contains all components of the robot necessary for driving.
+ */
+public class SparkDriveSubsystem extends Subsystem {
+	
+	private static SparkDriveSubsystem instance;
+	
+	static {
+		instance = new SparkDriveSubsystem();
+	}
+	
+	/**
+	 * Gets the current instance.
+	 * @return current instance of DriveSubsystem
+	 */
+	public static SparkDriveSubsystem getInstance() {
+		return instance;
+	}
+	
+	CHS_SparkMax leftSpark;
+	CHS_SparkMax rightSpark;
+	
+	DifferentialDrive teleopDrive;
+	
+	private HashMap<Double, Double> leftTable = new HashMap<>();
+	private HashMap<Double, Double> rightTable = new HashMap<>();
+	
+	/**
+	 * minimum power for which lookup table values have been determined
+	 */
+	private double minTestedPower;
+	
+	/**
+	 * maximum power for which lookup table values have been determined
+	 */
+	private double maxTestedPower;
+	
+	/*
+	 * status of encoder reset
+	 */
+	private boolean encoderResetComplete;
+	
+	/**
+	 * Initializes teleopDrive, motors, and lookup tables.
+	 */
+	private SparkDriveSubsystem() {
+		leftSpark = new CHS_SparkMax(RobotMap.SPARK_L, MotorType.kBrushless);
+		rightSpark = new CHS_SparkMax(RobotMap.SPARK_R, MotorType.kBrushless);
+
+		SpeedControllerGroup right = new SpeedControllerGroup(leftSpark.getSparkMaxObject());
+		SpeedControllerGroup left = new SpeedControllerGroup(rightSpark.getSparkMaxObject());
+		teleopDrive = new DifferentialDrive(right, left);
+		
+		initLookupTable();
+		
+		encoderResetComplete = false;
+	}
+	
+	/**
+	 * Initializes the motor power lookup tables. Inputs are raw motor powers, outputs are values that the raw powers will
+	 * be divided by.
+	 */
+	private void initLookupTable() {
+		leftTable.put(0.2, 1.035842);
+		leftTable.put(0.3, 1.033883);
+		leftTable.put(0.4, 1.044270);
+		leftTable.put(0.5, 1.055146);
+		leftTable.put(0.6, 1.052553);
+		leftTable.put(0.7, 1.061291);
+		leftTable.put(0.8, 1.088878);
+		
+		rightTable.put(0.2, 1.012183);
+		rightTable.put(0.3, 1.017407);
+		rightTable.put(0.4, 1.021134);
+		rightTable.put(0.5, 1.027404); // was 1.025404
+		rightTable.put(0.6, 1.020292);
+		rightTable.put(0.7, 1.017890);
+		rightTable.put(0.8, 1.033552);
+		
+		minTestedPower = Collections.min(leftTable.keySet());
+		maxTestedPower = Collections.max(leftTable.keySet());
+	}
+	
+	/**
+	 * Transforms raw motor powers using an experimental lookup table to account for the difference in motor outputs.
+	 * @param power		raw motor power
+	 * @param motor		motor for which the power is being set	
+	 * @return converted motor power
+	 */
+	/*public double convertPower(double power, WPI_TalonSRX motor) {
```

Comment:
It's generally a bad idea to comment out code as it may no longer compile. For instance, this is still referencing WPI_TalonSRX